rust/src/applayerpgsql/parser.rs

```diff
@@ -0,0 +1,597 @@
+        >> proto_minor: verify!(be_u16, |&x| x == PGSQL_DUMMY_PROTO_MINOR_SSL )
+        >> (PgsqlGenericStartupMessage {
+            message_type: PgsqlStartupMessageType::SslRequest,
+            length: len,
+            proto_major: proto_major,
+            proto_minor: proto_minor,
+        })
+));
+ 
+#[derive(Debug, PartialEq)]
+pub struct PgsqlSslResponse {
+    message_type: PgsqlSslResponseType,
+}
+
+impl fmt::Display for PgsqlSslResponse {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{:?}", self)
+    } 
+}
+
+named!(pub parse_ssl_response<PgsqlSslResponse>,
+    do_parse!(
+        identifier: alt!(char!('N') | char!('S'))
+        >> (PgsqlSslResponse{
+            message_type: PgsqlSslResponseType::from(identifier),
+        })
+    ));
+#[derive(Clone, Debug, PartialEq)]
+pub enum PgsqlErrorNoticeFieldTypes {
+    SeverityLocalizable,
+    SeverityNonLocalizable,
+    CodeSqlStateCode,
+    Message,
+    Detail,
+    Hint,
+    Position,
+    InternalPosition,
+    InternalQuery,
+    Where,
+    SchemaName,
+    TableName,
+    ColumnName,
+    DataType,
+    ConstraintName,
+    File,
+    Line,
+    Routine,
+    TerminatorToken,
+    UnknownFieldType,
+}
+
+impl From<char> for PgsqlErrorNoticeFieldTypes {
+    fn from(identifier: char) -> PgsqlErrorNoticeFieldTypes {
+        match identifier {
+            'S' => PgsqlErrorNoticeFieldTypes::SeverityLocalizable,
+            'V' => PgsqlErrorNoticeFieldTypes::SeverityNonLocalizable,
+            'C' => PgsqlErrorNoticeFieldTypes::CodeSqlStateCode,
+            'M' => PgsqlErrorNoticeFieldTypes::Message,
+            'D' => PgsqlErrorNoticeFieldTypes::Detail,
+            'H' => PgsqlErrorNoticeFieldTypes::Hint,
+            'P' => PgsqlErrorNoticeFieldTypes::Position,
+            'p' => PgsqlErrorNoticeFieldTypes::InternalPosition,
+            'q' => PgsqlErrorNoticeFieldTypes::InternalQuery,
+            'W' => PgsqlErrorNoticeFieldTypes::Where,
+            's' => PgsqlErrorNoticeFieldTypes::SchemaName,
+            't' => PgsqlErrorNoticeFieldTypes::TableName,
+            'c' => PgsqlErrorNoticeFieldTypes::ColumnName,
+            'd' => PgsqlErrorNoticeFieldTypes::DataType,
+            'n' => PgsqlErrorNoticeFieldTypes::ConstraintName,
+            'F' => PgsqlErrorNoticeFieldTypes::File,
+            'L' => PgsqlErrorNoticeFieldTypes::Line,
+            'R' => PgsqlErrorNoticeFieldTypes::Routine,
+            '\u{0}' => PgsqlErrorNoticeFieldTypes::TerminatorToken,
+            _ => PgsqlErrorNoticeFieldTypes::UnknownFieldType, // adding this because documentation says "rontends should silently ignore fields of unrecognized type."
+        }
+    }
+}
+
+impl From<u8> for PgsqlErrorNoticeFieldTypes {
+    fn from(identifier: u8) -> PgsqlErrorNoticeFieldTypes {
+        match identifier {
+            0x53 => PgsqlErrorNoticeFieldTypes::SeverityLocalizable,
```

Comment:
Not sure how, but I think Rust will allow us to use the human readable versions here. Was it `b'A'` or something?